OpenLaTeXStudio/EditorModule/src/latexstudio/editor/EditorTopComponent.java

```diff
@@ -291,12 +331,60 @@ void writeProperties(java.util.Properties p) {
         // http://wiki.apidesign.org/wiki/PropertyFiles
         p.setProperty("version", "1.0");
         // TODO store your settings
-    }
-
+        if(currentFile != null) {
+           p.setProperty(CURRENT_FILE_KEY, currentFile.getAbsolutePath());
+        }
+        else {
+            p.remove(CURRENT_FILE_KEY);
+        }
+        if(latexPath != null) {
+           p.setProperty(LATEX_PATH_KEY, latexPath);
+        }
+        else {
+            p.remove(LATEX_PATH_KEY);
+        }
+        PropertyService.writeProperties(p);
+    } 
+   //previous argumnent: java.util.Properties p
     void readProperties(java.util.Properties p) {
-        String version = p.getProperty("version");
         // TODO read your settings according to their version
+        String currentFilePath;
+        File temp;
+        try {
+           p = PropertyService.readProperties(PropertyService.PROPERTIES_FILEPATH);
+        }
+        catch(NullPointerException exception) {
+           return;
+        }
+        try {
+        currentFilePath = (String) p.get(CURRENT_FILE_KEY);
+        
+        temp = new File(currentFilePath);
+        if(temp.exists() && temp.canRead()) 
+            {
+                this.setCurrentFile(temp);
+            }
+        }
+         catch(NullPointerException exception) {
```

Comment:
catching NullPointerExceptions is considered a bad practice, since they can easily be avoided by simple null checks - if (object != null)